src/main/java/com/github/manolo8/darkbot/gui/players/PlayerEditor.java

```diff
@@ -4,29 +4,71 @@
 import com.github.manolo8.darkbot.config.ConfigEntity;
 import com.github.manolo8.darkbot.config.PlayerInfo;
 import com.github.manolo8.darkbot.config.PlayerTag;
-import com.github.manolo8.darkbot.config.UnresolvedPlayer;
-import com.github.manolo8.darkbot.gui.components.MainButton;
 import com.github.manolo8.darkbot.gui.utils.Popups;
 import com.github.manolo8.darkbot.gui.utils.SearchField;
-import com.github.manolo8.darkbot.gui.utils.UIUtils;
 import com.github.manolo8.darkbot.utils.I18n;
+import eu.darkbot.api.events.EventHandler;
+import eu.darkbot.api.events.Listener;
+import eu.darkbot.api.game.entities.Player;
+import eu.darkbot.api.managers.EntitiesAPI;
+import eu.darkbot.api.managers.EventBrokerAPI;
 import net.miginfocom.swing.MigLayout;
 
 import javax.swing.*;
-import java.awt.event.ActionEvent;
-import java.util.Comparator;
-import java.util.List;
-import java.util.Locale;
+import java.util.*;
 
-public class PlayerEditor extends JPanel {
+import static java.util.stream.Collectors.toList;
+
+public class PlayerEditor extends JPanel implements Listener {
     private final JList<PlayerInfo> playerInfoList;
     private final DefaultListModel<PlayerInfo> playersModel;
+    private final List<PlayerInfo> nearbyPlayerList = new ArrayList<>();
 
     private final PlayerManager playerManager;
     private final SearchField sf;
     private final PlayerTagManager tagEditor;
     protected Main main;
 
+    @EventHandler
+    public void onEntityPlayerAdd(EntitiesAPI.EntityCreateEvent event) {
+        if (event.getEntity() instanceof Player) {
+            if (playersModel != null && main.config.PLAYER_INFOS != null && main.config.PLAYER_INFOS.values()
+                    .stream().noneMatch(a -> a.toString()
+                            .contains(((Player) event.getEntity()).getEntityInfo().getUsername()))) {
+                nearbyPlayerList.add(new PlayerInfo(((Player) event.getEntity())
+                        .getEntityInfo().getUsername(), event.getEntity().getId()));
+
+                SwingUtilities.invokeLater(() -> playersModel.
+                        addElement(new PlayerInfo(((Player) event.getEntity())
+                                .getEntityInfo().getUsername(), event.getEntity().getId())));
```

Comment:
Just add a constructor to PlayerInfo that takes in a Player as parameter to simplify this

```suggestion
                        addElement(new PlayerInfo((Player) event.getEntity()));
```